### PR TITLE
tf_not_equal fix

### LINKED ISSUE #15615: Fix math.test_tensorflow_not_equal
|   |   |
|---|---|
|numpy|<a href="https://github.com/unifyai/ivy/actions/runs/6272580647/job/17034377037"></a>
|jax|<a href="https://github.com/unifyai/ivy/actions/runs/6272580647/job/17034377037"></a>
|tensorflow|<a href="https://github.com/unifyai/ivy/actions/runs/6272580647/job/17034377037"></a>
|torch|<a href="https://github.com/unifyai/ivy/actions/runs/6272580647/job/17034377037"></a>
|paddle|<a href="https://github.com/unifyai/ivy/actions/runs/6272580647/job/17034377037"></a>

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #15615

### CODE REVIEW

**Comment by mobley-trent** [comment]
The paddle backend for `not_equal` only supports float32, float64, int32 and int64. This is what caused the test failure for paddle

**Comment by mobley-trent** [comment]
@hirwa-nshuti

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -481,6 +481,10 @@ def equal(x, y, name=None):
     return ivy.equal(x, y)
 
 
+@with_supported_dtypes(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/tensorflow/math.py:484]: I think these are not only supported dtypes 😅
```python
import tensorflow as tf
print(tf.math.not_equal(tf.constant([-1.2+1.2j, 0.0+0.0j, 1.5+1.5j]), tf.constant([-1.2+1.2j, 0.0+0.0j, 1.5+1.5j])))
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @mobley-trent 

Would you check thoroughly all supported dtypes

File: ivy/functional/frontends/tensorflow/math.py [hunk 2]
```diff
@@ -481,6 +481,10 @@ def equal(x, y, name=None):
...
+    {"2.9.0 and below": ("float32", "float64", "int32", "int64")},
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/tensorflow/math.py:485]: The version is 2.12.0

**Comment by mobley-trent** [comment]
@hirwa-nshuti I set the valid dtypes in the test and removed the decorator entirely

**Comment by hirwa-nshuti** [comment]
Hey @mobley-trent can you try to update the branch so that the CI check should run

**Pull request merged**